### PR TITLE
Feature order edit

### PR DESCRIPTION
**Pull request by MicheleRezk**
- Adding OrderEdits Domain Models
- Adding OrderEdits Commands, Request Builder
- Adding OrderEdits Integration Tests
- Adding InStoreCommand as decorator for other commands and applying it for Customers, Carts and Orders endpoint
- Adding the store field to the OrderImportDraft
- Adding JsonConverter to support Deserialization of IReference types
- Create TokenSerializerService with SnakeCaseNamingStrategy and update token providers to use this new service
- Rename Order Update Action "SetShipmentReturnStateUpdateAction" to be "SetReturnShipmentStateUpdateAction"
- Rename folder of Registration Project from "commercetools.Sdk.Reflection" to "commercetools.Sdk.Registration"
- Rename "HttpApiErrorResponse" class to "ErrorResponse"
- Move these classes (ErrorResponse, Error, GeneralError, ConcurrentModificationError) from "commercetools.Sdk.HttpApi.Domain" project to "commercetools.Sdk.Domain" project

### CODE REVIEW

File: commercetools.Sdk/IntegrationTests/commercetools.Sdk.IntegrationTests/Carts/CartsFixture.cs [hunk 1]
```diff
@@ -38,6 +40,13 @@ public static CartDraft DefaultCartDraft(CartDraft cartDraft)
             return cartDraft;
         }
 
+        public static CartDraft DefaultCartDraftInStore(CartDraft cartDraft, ResourceIdentifier<Store> store)
```
Comment (jenschude) [inline commercetools.Sdk/IntegrationTests/commercetools.Sdk.IntegrationTests/Carts/CartsFixture.cs]: Could this be changed to a IReferenceable?

Comment (MicheleRezk) [reply commercetools.Sdk/IntegrationTests/commercetools.Sdk.IntegrationTests/Carts/CartsFixture.cs]: yes, it's solved now :thumbsup:

File: commercetools.Sdk/commercetools.Sdk.HttpApi/RequestBuilders/GetMatchingQueryRequestMessageBuilder.cs [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+            queryStringParameters.ForEach(x => { requestUri = QueryHelpers.AddQueryString(requestUri, x.Key, x.Value); });
+            return new Uri(requestUri, UriKind.Relative);
+        }
+
+        private string GetMatchingUriSuffix<T>(GetMatchingQueryCommand<T> command)
+        {
+            string suffix;
+            switch (command)
+            {
+                case GetShippingMethodsForCartCommand shippingMethodsForCartCommand:
```
Comment (jenschude) [inline commercetools.Sdk/commercetools.Sdk.HttpApi/RequestBuilders/GetMatchingQueryRequestMessageBuilder.cs]: You could add a property or attribute to the GetMatchingQueryCommand to retrieve the suffix

Comment (MicheleRezk) [reply commercetools.Sdk/commercetools.Sdk.HttpApi/RequestBuilders/GetMatchingQueryRequestMessageBuilder.cs]: Done :thumbsup:

File: commercetools.Sdk/commercetools.Sdk.Client/GetMatchingQueryCommand.cs [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+        protected GetMatchingQueryCommand(IAdditionalParameters<T> additionalParameters)
+        {
+            this.AdditionalParameters = additionalParameters;
+        }
+
+        public override System.Type ResourceType => typeof(T);
+
+        public List<string> Expand { get; set; }
+
+        public string UrlSuffix { get; set; }
```
Comment (jenschude) [inline commercetools.Sdk/commercetools.Sdk.Client/GetMatchingQueryCommand.cs]: Maybe remove the set?

`public virtual string UrlSuffix => string.Empty`

Comment (MicheleRezk) [reply commercetools.Sdk/commercetools.Sdk.Client/GetMatchingQueryCommand.cs]: :thumbsup:

**Pull request merged**